CONTRIBUTING.md

```diff
@@ -1,108 +1,170 @@
+
+To publish a release manually, from the current local code:
+1. Create a tag (e.g. `test123`) and push it to the remote repository. The tag should not include any uppercase letters.
     ```
-    > ./gradlew buildDockerImage
-    [...Docker build output...]
-    terra-cli/local:b5fdce0 successfully built
-    
-    BUILD SUCCESSFUL in 25s
-    1 actionable task: 1 executed
+    > git tag -a test123 -m "testing version 123"
+    > git push --tags
     ```
-2. Update the image id that the CLI uses. (See output of previous command for image name and tag.)
+2. Update the version in `build.gradle`.
     ```
-    > terra app set-image --image=terra-cli/local:b5fdce0
+    version = 'test123'
+    ```
+3. Login to GitHub and run the `tools/publish-release.sh` script. This will publish a pre-release, which does not
+affect the "Latest release" tag.
+    ```
+    > gh auth login
+    > ./tools/publish-release.sh test123
     ```
+    To publish a regular release, add `true` as a second argument.
+     ```
+    > gh auth login
+    > ./tools/publish-release.sh test123 true
+    ```
+
+Note that GitHub automatically attaches an archive of the source code to the release. If you have local changes that
+are not yet committed, then they may not be reflected in the source code archive, but they will be included in the
+install package. We don't use the source code archive for install.
+
+#### Install overview
+Three shell scripts are published for users.
+
+##### `download-install.sh`
+This is convenience script that:
+- Downloads the latest (or specific version) of the install package
+- Unarchives it
+- Runs the `install.sh` script included inside
+- Deletes the install package
+
+It is published as a separate file in each GitHub release.
+The intent is to have a one-line install command  `curl -L download-install.sh | bash`.
+
+##### `install.sh`
+This is an installer script that:
+- Moves all the JARs to `$HOME/.terra/lib`
+- Moves the `terra` run script and the `README.md` file outside the unarchived install package directory
+- Deletes the unarchived install package directory
+- Sets the Docker image id to the default
+- Pulls the default Docker image id
+
+It is included in the `terra-cli.tar` install package in each GitHub release.
+It needs to be run from the same directory: `./install.sh`
+
+##### `terra`
+This is the run script that wraps the Java call to the CLI.
+- It looks for the JARs on the classpath in the `$HOME/.terra/lib` directory.
+- This script is generated by the Gradle application plugin, so any changes should be made there.
+
+It is included in the `terra-cli.tar` install package in each GitHub release.
+This is the script users can add to their `$PATH` to invoke the CLI more easily from another directory.
+
+
+### Docker
+The `docker/` directory contains files required to build the Docker image.
+All files in the `scripts/` sub-directory are copied to the image, into a sub-directory that is on the `$PATH`, 
+and made executable.
```

Comment:
Can the installer also put the `terra` script on the path?

Reply:
Bash scripts can't modify the environment variables in the calling process, unless you use the `source` command. So, we can't really do this nicely. This pattern of installing in the current directory is also common to tools like Nextflow, so I think it makes sense to imitate here.